tsdb/querier.go

```diff
@@ -661,7 +684,7 @@ func (p *populateWithDelSeriesIterator) Err() error {
 }
 
 type populateWithDelChunkSeriesIterator struct {
-	*populateWithDelGenericSeriesIterator
+	populateWithDelGenericSeriesIterator
```

Comment:
Are we sure this is safe?

Reply:
I'm not sure what worries you.
In this implementation, `populateWithDelGenericSeriesIterator` is a "base class", only Go doesn't have inheritance so it is an anonymous field embedded in the outer struct.

Previously the code was setting up an extra object on the heap; now it is embedded so we save an allocation.